modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -2036,7 +2091,128 @@ public Set<String> getDataNodesByZoneId(int zoneId) {
         return inBusyLock(busyLock, () -> {
             ZoneState zoneState = zonesState.get(zoneId);
 
-            return zoneState.nodes;
+            return zoneState.nodes();
         });
     }
+
+    /**
+     * Returns local mapping of {@code nodeId} -> node's attributes, where {@code nodeId} is a node id, that changes between restarts.
+     * This map is updated every time we receive a topology event in a {@code topologyWatchListener}.
+     *
+     * @return Mapping {@code nodeId} -> node's attributes.
+     */
+    public Map<String, Map<String, String>> nodesAttributes() {
+        return nodesAttributes;
+    }
+
+    @TestOnly
+    Map<Integer, ZoneState> zonesTimers() {
+        return zonesState;
+    }
+
+    /**
+     * Structure that represents node with the attributes and which we store in Meta Storage when we store logical topology.
+     * Light-weighted version of the {@link LogicalNode}.
+     */
+    public static class NodeWithAttributes implements Serializable {
```

Comment:
- serialVersionUID for both Node and NodeWithAttributes
- I'd rather extract given classes, because they are intentionally made public and are used not only wihtin DZM.